### ISSUE TITLE
Bug: Attribute Search internal server error

### ISSUE THREAD

**Issue by Q-Feeds** [MISP/MISP#10035]
### Actual behavior

When searching for an attribute within the GUI: 

An Internal Error Has Occurred.
Error: An Internal Error Has Occurred. Please try your action again. If the problem persists, please contact administrator.

### Expected behavior

Obviously showing the search results

### Steps to reproduce

Search for an attribute in the GUI. (no filters applied)

### Version

v2.5.1 (3077ed2cec7c0a703f2aaeb1f726e0199e067ae3)

### Operating System

Ubuntu

### Operating System version

Ubuntu 24.04.1 LTS (GNU/Linux 6.8.0-48-generic x86_64)

### PHP version

8.3.6 

### Browser

Tested on Safari and Chrome

### Relevant log output

```shell
2024-11-10 17:33:56 Error: [Error] Call to a member function query() on null
Request URL: /attributes/search/results
Stack Trace:
#0 [internal function]: AttributesController->search()
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(500): ReflectionMethod->invokeArgs()
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#4 /var/www/MISP/app/webroot/index.php(107): Dispatcher->dispatch()
#5 {main}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aBUDmdBQ** [comment]
### The same Problem in MISP-Version 2.4.X
When searching for an attribute within the GUI:

An Internal Error Has Occurred.
Error: An Internal Error Has Occurred. Please try your action again. If the problem persists, please contact administrator.
Expected behavior



### Steps to reproduce

Search for an attribute in the GUI. (no filters applied)

### Version
v2.4.199 (https://github.com/MISP/MISP/releases/tag/v2.4.199)

### Operating System
Ubuntu

### Operating System version

Ubuntu 22.04.1 LTS (GNU/Linux 5.15.0-97-generic x86_64)
### PHP version

7.4.33 

### Relevant log output

```
2024-11-11 16:46:10 Error: [Error] Call to a member function query() on null
Request URL: /attributes/search/results
Stack Trace:
#0 [internal function]: AttributesController->search()
#1 /srv/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(499): ReflectionMethod->invokeArgs()
#2 /srv/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#3 /srv/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#4 /srv/www/MISP/app/webroot/index.php(105): Dispatcher->dispatch()
#5 {main}
```

**Comment by HiS3** [comment]
### The same Problem in MISP-Version 2.4.X
When searching for an attribute within the GUI:

An Internal Error Has Occurred.
Error: An Internal Error Has Occurred. Please try your action again. If the problem persists, please contact administrator.
Expected behavior

### Steps to reproduce

Search for an attribute in the GUI. (no filters applied)

### Version
v2.4.199 (https://github.com/MISP/MISP/releases/tag/v2.4.199)

### Operating System
Ubuntu

### Operating System version

Ubuntu 22.04.1 LTS (GNU/Linux 5.15.0-97-generic x86_64)
### PHP version

7.4.33 

### Relevant log output

```
2024-11-11 16:46:10 Error: [Error] Call to a member function query() on null
Request URL: /attributes/search/results
Stack Trace:
#0 [internal function]: AttributesController->search()
#1 /srv/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(499): ReflectionMethod->invokeArgs()
#2 /srv/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction()
#3 /srv/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke()
#4 /srv/www/MISP/app/webroot/index.php(105): Dispatcher->dispatch()
#5 {main}
```

**Comment by marjatech** [comment]
For 2.4 this was already fixed in 157966c 
However this is not in v2.4.199

**Comment by iglocska** [comment]
We'll do a release today and it will include the fix, sit tight folks!

**Comment by Q-Feeds** [comment]
> We'll do a release today and it will include the fix, sit tight folks!

I assume it’s postponed ?

**Comment by Ghiyarandhika** [comment]
is it already fix? @Q-Feeds

**Comment by Q-Feeds** [comment]
> is it already fix? @Q-Feeds

Still waiting for a release I’m afraid.

**Comment by Ghiyarandhika** [comment]
i see, tag pls me when its release @Q-Feeds

**Comment by trizzosk** [comment]
> For 2.4 this was already fixed in [157966c](https://github.com/MISP/MISP/commit/157966cbaa1edf8710f1157f3a0f5f17d0d68da2) However this is not in v2.4.199

I have 2.4.199 and I was affected. I manually added fix from mentioned commit and it solved my problem.

**Comment by gokulsuresh01** [comment]
is there a fix for this yet?

**Comment by trizzosk** [comment]
@gokulsuresh01 take a look at the [157966c](https://github.com/MISP/MISP/commit/157966cbaa1edf8710f1157f3a0f5f17d0d68da2). It worked for me.

I did a manual adjustment in code, leaving original file as backup (in case there will be another 2.4.x release).

**Comment by Q-Feeds** [comment]
@Ghiyarandhika it's' fixed in the latest 2.5 release :D

**Comment by Ghiyarandhika** [comment]
OK Cool
